evalml/automl/engine/engine_base.py

```diff
@@ -157,7 +158,11 @@ def train_pipeline(pipeline, X, y, automl_config, schema=True):
         X,
         y,
     )
-    return cv_pipeline
+    if not get_hashes:
+        return cv_pipeline
+
+    X_hash = hash(tuple(X.index))
```

Comment:
Right, the hash here is now mainly to reduce the memory consumption. If we are passing in data with lots of rows, keeping references to all of the rows for each cv split could end up consuming a fair bit of memory. We can simplify this by hashing, and this hashing call itself is fairly quick, which allows us to easily and conveniently use this.

Will update the docstring though!

Reply:
Will it consume a lot of memory?  The index is basically just an array of integers.

Reply:
Mmm, I think it could be substantial if the dataframe passed in is large, ie 100k+ rows.

Python's hashing algo is quite fast, so I feel comfortable relying on this to control our memory usage.